pnc-rest/src/main/java/org/jboss/pnc/rest/provider/ProductProvider.java

```diff
@@ -24,8 +24,23 @@ public ProductProvider(ProductRepository productRepository) {
     public ProductProvider() {
     }
 
-    public List<ProductRest> getAll() {
-        return productRepository.findAll().stream().map(product -> new ProductRest(product)).collect(Collectors.toList());
+    @Override
+    public Function<? super Product, ? extends ProductRest> mapper() {
+        return product -> new ProductRest(product);
+    }
+
+    @Override
+    public String getDefaultSortingField() {
+        return "name";
+    }
+
+    public Object getAll(Integer pageIndex, Integer pageSize, String field, String sorting) {
+
+        if (noPaginationRequired(pageIndex, pageSize, field, sorting)) {
+            return productRepository.findAll().stream().map(mapper()).collect(Collectors.toList());
+        } else {
+            return transform(productRepository.findAll(buildPageRequest(pageIndex, pageSize, field, sorting)));
```

Comment:
ok as clarified on IRC, a PageRequest is already fetched into the findAll() method, so Hibernate is doing the pagination work for us already.